### DISCUSSION TITLE
Sonoff ZbBridgePro not working with ZHA Home Assistant

### DISCUSSION THREAD

**Discussion by tzago** [arendst/Tasmota#20924] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
Sonoff ZbBridgePro not working with ZHA Home Assistant
Latest version deployed in my Sonoff ZigbeePro is Tasmota 13.4.0(zbbrdgpro) by Theo Arends Installed from Web Installer

Following the instructions from here to connect my ZHA to my Sonoff Tasmoted to 13.4.0(zbbrdgpro) by Theo Arends
https://zigbee.blakadder.com/Sonoff_ZBBridge-P.html

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ X] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [X ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ X] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ X] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ X] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ X] Device used (e.g., Sonoff Basic): Sonoff ZigbeePro
- [ X] Tasmota binary firmware version number used:  13.4.0(zbbrdgpro) by Theo Arends 
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ X] Flashing tools used: Web installer
- [X  ] Provide the output of command: `Backlog Template; Module; GPIO 255`:

```lua
  Configuration output here:

#Console run: Backlog Template; Module; GPIO 255
05:12:41.927 RSL: RESULT = {"NAME":"TCP ZBBridge Pro","GPIO":[0,0,576,0,480,0,0,0,0,1,1,5792,0,0,0,5472,0,320,5793,5504,0,640,608,32,0,0,0,0,0,1,0,0,0,0,0,0],"FLAG":0,"BASE":1}
05:12:42.152 RSL: RESULT = {"Module":{"0":"TCP ZBBridge Pro"}}
05:12:42.409 RSL: RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"576":"LedLink_i"},"GPIO3":{"0":"None"},"GPIO4":{"480":"Buzzer"},"GPIO5":{"0":"None"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"5792":"Zigbee Rst1"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"5472":"TCP Tx"},"GPIO20":{"0":"None"},"GPIO21":{"320":"Led_i1"},"GPIO22":{"5793":"Zigbee Rst2"},"GPIO23":{"5504":"TCP Rx"},"GPIO24":{"0":"None"},"GPIO25":{"640":"I2C SDA1"},"GPIO26":{"608":"I2C SCL1"},"GPIO27":{"32":"Button1"},"GPIO32":{"0":"None"},"GPIO33":{"0":"None"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO37":{"0":"None"},"GPIO38":{"0":"None"},"GPIO39":{"0":"None"}}

#Information button
TCP ZBBridge Pro
Tasmota
Program Version	13.4.0(zbbrdgpro)
Build Date & Time	2024-02-14T16:17:21
Core/SDK Version	2_0_14/4.4.6.240105
Uptime	0T12:04:46
Flash Write Count	16
Boot Count	6
Restart Reason	Software reset CPU
AP1 Information	SSId TMO555
RSSI 100%, -43 dBm
Mode 11n
Channel 6
BSSId 75:55:44:13:CC:DD
Hostname	tasmota-2D6E2C-3628
IPv6 Local (WiFi)	fe80::a3d:55dd:fe5c:3d5f%st1
MAC Address	0C:8B:95:2D:6E:2C
IP Address (WiFi)	192.168.0.5
Gateway	192.168.0.1
Subnet Mask	255.255.255.0
DNS Server1	192.168.0.1
DNS Server2	0.0.0.0
HTTP API	Enabled
MQTT Host	
MQTT Port	1883
MQTT TLS	Disabled
MQTT User	DVES_USER
MQTT Client	DVES_2D6E2C
MQTT Topic	tasmota_%06X
MQTT Group Topic 1	cmnd/tasmotas/
MQTT Full Topic	cmnd/tasmota_2D6E2C/
MQTT Fallback Topic	cmnd/DVES_2D6E2C_fb/
MQTT No Retain	Disabled
Emulation	None
ESP Chip Id	2977324 (ESP32-D0WD-V3 v3.0)
Flash Chip Id	0x16405E (DIO)
Flash Size	4096 KB
Program Flash Size	4096 KB
Program Size	1477 KB
Free Program Space	378 KB
Free Memory	153.6 KB (frag. 28%)
Partition safeboot	832 KB (used 93%)
Partition app0*	1856 KB (used 79%)
Partition fs	1344 KB

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
with Sonoff Zigbee Pro Autoconfigured.
#Both  "Zigbee Permit" Join and "Zigbee Map buttons" are disabled.

00:00:00.003 HDW: ESP32-D0WD-V3 v3.0 
00:00:00.095 UFS: FlashFS mounted with 908 kB free
00:00:00.113 CFG: Loaded from File, Count 13
00:00:00.115 FRC: Some settings have been reset (2)
00:00:00.120 I2C: Bus1 using GPIO26(SCL) and GPIO25(SDA)
00:00:00.204 BRY: Berry initialized, RAM used 4019 bytes
00:00:00.251 Project tasmota - Tasmota Version 13.4.0(zbbrdgpro)-2_0_14(2024-02-14T16:17:21)
00:00:02.000 WIF: Connecting to AP1 TMO555 Channel 6 BSSId 25:55:99:24:FF:01 in mode 11n as tasmota-2D6E2C-3628...
00:00:03.587 WIF: Connected
00:00:03.895 HTP: Web server active on tasmota-2D6E2C-3628 with IP address 192.168.0.5
16:57:08.100 RSL: INFO1 = {"Info1":{"Module":"Sonoff Zigbee Pro","Version":"13.4.0(zbbrdgpro)","FallbackTopic":"cmnd/DVES_2D6E2C_fb/","GroupTopic":"cmnd/tasmotas/"}}
16:57:08.116 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-2D6E2C-3628","IPAddress":"192.168.0.5","IP6Global":"","IP6Local":"fe80::a3d:55dd:fe5c:3d5f%st1"}}
16:57:08.130 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":5}}
16:57:08.153 RUL: SYSTEM#BOOT performs "TCPStart 8888"
16:57:08.159 RSL: RESULT = {"Command":"Unknown"}
16:57:10.907 QPC: Reset
16:57:12.906 RSL: STATE = {"Time":"2024-03-10T16:57:12","Uptime":"0T00:00:08","UptimeSec":8,"Heap":160,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":38},"Wifi":{"AP":1,"SSId":"TMO555","BSSId":"75:55:44:13:CC:DD","Channel":6,"Mode":"11n","RSSI":100,"Signal":-40,"LinkCount":1,"Downtime":"0T00:00:03"}}
16:57:18.651 ZIG: rebooting ZNP device
16:57:20.701 ZbInput discarding byte 00
16:57:20.704 RSL: RESULT = {"ZbState":{"Status":1,"Message":"CCxxxx ZNP booted","RestartReason":"Power-up","MajorRel":2,"MinorRel":7}}
16:57:20.903 RSL: RESULT = {"ZbState":{"Status":50,"MajorRel":2,"MinorRel":7,"MaintRel":19,"Revision":0}}
16:57:21.053 RSL: RESULT = {"ZbState":{"Status":2,"Message":"Resetting configuration"}}
16:57:23.601 ZbInput discarding byte 00
16:57:24.153 RSL: RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
16:57:24.157 ZIG: Abort
16:57:24.158 ZIG: Stopping (99)
16:58:15.178 TCP: Got connection from 192.168.0.9
16:58:28.954 TCP: Got connection from 192.168.0.9

#TCP ZBBridge Pro TCP auto configured...
#Console
00:00:00.003 HDW: ESP32-D0WD-V3 v3.0 
00:00:00.071 UFS: FlashFS mounted with 908 kB free
00:00:00.085 CFG: Loaded from File, Count 15
00:00:00.087 FRC: Some settings have been reset (2)
00:00:00.093 I2C: Bus1 using GPIO26(SCL) and GPIO25(SDA)
00:00:00.160 BRY: Berry initialized, RAM used 4023 bytes
00:00:00.197 Project tasmota - Tasmota Version 13.4.0(zbbrdgpro)-2_0_14(2024-02-14T16:17:21)
00:00:02.001 WIF: Connecting to AP1 TMO555 Channel 6 BSSId 75:55:44:13:CC:DD in mode 11n as tasmota-2D6E2C-3628...
00:00:03.694 WIF: Connected
00:00:03.902 HTP: Web server active on tasmota-2D6E2C-3628 with IP address 192.168.0.5
17:01:54.155 RSL: INFO1 = {"Info1":{"Module":"TCP ZBBridge Pro","Version":"13.4.0(zbbrdgpro)","FallbackTopic":"cmnd/DVES_2D6E2C_fb/","GroupTopic":"cmnd/tasmotas/"}}
17:01:54.171 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-2D6E2C-3628","IPAddress":"192.168.0.5","IP6Global":"","IP6Local":"fe80::a3d:55dd:fe5c:3d5f%st1"}}
17:01:54.185 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":6}}
17:01:54.207 RUL: SYSTEM#BOOT performs "TCPStart 8888"
17:01:54.210 TCP: Starting TCP server on port 8888
17:01:54.213 RSL: RESULT = {"TCPStart":"Done"}
17:01:56.911 QPC: Reset
17:01:58.910 RSL: STATE = {"Time":"2024-03-10T17:01:58","Uptime":"0T00:00:08","UptimeSec":8,"Heap":161,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":45},"Wifi":{"AP":1,"SSId":"TMO555","BSSId":"75:55:44:13:CC:DD","Channel":6,"Mode":"11n","RSSI":100,"Signal":-36,"LinkCount":1,"Downtime":"0T00:00:03"}}
17:02:43.225 TCP: Got connection from 192.168.0.9
17:02:43.988 TCP: Got connection from 192.168.0.9

Checking ZHA logs I can see.

File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 72, in wrapper
result = await method(view, request, data, *args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 111, in post
result = await self._flow_mgr.async_configure(flow_id, data)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 359, in async_configure
result = await self._async_configure(flow_id, user_input)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 401, in _async_configure
result = await self._async_handle_step(
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 502, in _async_handle_step
result: FlowResult = await getattr(flow, method)(user_input)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/zha/config_flow.py", line 382, in async_step_form_initial_network
return await self.async_step_form_new_network(user_input)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/zha/config_flow.py", line 388, in async_step_form_new_network
await self._radio_mgr.async_form_network()
File "/usr/src/homeassistant/homeassistant/components/zha/radio_manager.py", line 276, in async_form_network
await app.form_network()
File "/usr/local/lib/python3.12/site-packages/zigpy/application.py", line 366, in form_network
await self.start_network()
File "/usr/local/lib/python3.12/site-packages/zigpy_znp/zigbee/application.py", line 177, in start_network
await self.load_network_info()
File "/usr/local/lib/python3.12/site-packages/zigpy_znp/zigbee/application.py", line 144, in load_network_info
await self._znp.load_network_info(load_devices=load_devices)
File "/usr/local/lib/python3.12/site-packages/zigpy_znp/api.py", line 238, in load_network_info
await self._load_network_info(load_devices=load_devices)
File "/usr/local/lib/python3.12/site-packages/zigpy_znp/api.py", line 142, in _load_network_info
key_desc = await self.nvram.osal_read(
^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/zigpy_znp/nvram.py", line 191, in osal_read
raise SecurityError(
zigpy_znp.exceptions.SecurityError: NV item cannot be read due to security constraints: <OsalNvIds.NWK_ACTIVE_KEY_INFO: 58>



```

### TO REPRODUCE
_Steps to reproduce the behavior:_
Latest version deployed in my Sonoff ZigbeePro is Tasmota 13.4.0(zbbrdgpro) by Theo Arends Installed from Web Installer
and tried to integrate with my ZHA using  https://zigbee.blakadder.com/Sonoff_ZBBridge-P.html as a reference

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
ZHA integrated with my Sonoff ZigbeePro is Tasmota 

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._
 

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by s-hadinger** [comment, marked as answer]
From what I can see, the TCP connection works. Use `Weblog 4` to see the actual traffic going through TCP.

The rest should be asked in ZHA forum. In this scenario Tasmota just passes opaque bytes between ZHA and the MCU

**Reply by tzago** [reply]
Enabling weblog 4 in the consoloe

`00:00:00.003-224/53 HDW: ESP32-D0WD-V3 v3.0 
00:00:00.100-223/52 UFS: FlashFS mounted with 908 kB free
00:00:00.119 CFG: Loaded from File, Count 16
00:00:00.129 QPC: Count 1
00:00:00.135 I2C: Bus1 using GPIO26(SCL) and GPIO25(SDA)
00:00:00.221 BRY: Berry initialized, RAM used 4023 bytes
00:00:00.268 Project tasmota - Tasmota Version 13.4.0(zbbrdgpro)-2_0_14(2024-02-14T16:17:21)
00:00:02.001 WIF: Connecting to AP1 TMO555 Channel 6 BSSId 75:55:44:13:CC:DD in mode 11n as tasmota-2D6E2C-3628...
00:00:03.643 WIF: Connected
00:00:03.902 HTP: Web server active on tasmota-2D6E2C-3628 with IP address 192.168.0.5
15:45:16.636 RSL: INFO1 = {"Info1":{"Module":"TCP ZBBridge Pro","Version":"13.4.0(zbbrdgpro)","FallbackTopic":"cmnd/DVES_2D6E2C_fb/","GroupTopic":"cmnd/tasmotas/"}}
15:45:16.652 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-2D6E2C-3628","IPAddress":"192.168.0.5","IP6Global":"","IP6Local":"fe80::a3d:55dd:fe5c:3d5f%st1"}}
15:45:16.666 RSL: INFO3 = {"Info3":{"RestartReason":"Vbat power on reset","BootCount":7}}
15:45:16.690 RUL: SYSTEM#BOOT performs "TCPStart 8888"
15:45:16.693 TCP: Starting TCP server on port 8888
15:45:16.696 RSL: RESULT = {"TCPStart":"Done"}
15:45:18.895 QPC: Reset
15:45:20.894 RSL: STATE = {"Time":"2024-03-11T15:45:20","Uptime":"0T00:00:09","UptimeSec":9,"Heap":163,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":45},"Wifi":{"AP":1,"SSId":"TMO555","BSSId":"75:55:44:13:CC:DD","Channel":6,"Mode":"11n","RSSI":100,"Signal":-34,"LinkCount":1,"Downtime":"0T00:00:03"}}
15:50:20.856 RSL: STATE = {"Time":"2024-03-11T15:50:20","Uptime":"0T00:05:09","UptimeSec":309,"Heap":163,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":45},"Wifi":{"AP":1,"SSId":"TMO555","BSSId":"75:55:44:13:CC:DD","Channel":6,"Mode":"11n","RSSI":100,"Signal":-35,"LinkCount":1,"Downtime":"0T00:00:03"}}
15:55:10.856 APP: Serial logging disabled
15:55:20.862 RSL: STATE = {"Time":"2024-03-11T15:55:20","Uptime":"0T00:10:09","UptimeSec":609,"Heap":163,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":45},"Wifi":{"AP":1,"SSId":"TMO555","BSSId":"75:55:44:13:CC:DD","Channel":6,"Mode":"11n","RSSI":100,"Signal":-35,"LinkCount":1,"Downtime":"0T00:00:03"}}
16:00:20.866 RSL: STATE = {"Time":"2024-03-11T16:00:20","Uptime":"0T00:15:09","UptimeSec":909,"Heap":163,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":45},"Wifi":{"AP":1,"SSId":"TMO555","BSSId":"75:55:44:13:CC:DD","Channel":6,"Mode":"11n","RSSI":100,"Signal":-34,"LinkCount":1,"Downtime":"0T00:00:03"}}
16:05:20.873 RSL: STATE = {"Time":"2024-03-11T16:05:20","Uptime":"0T00:20:09","UptimeSec":1209,"Heap":163,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":45},"Wifi":{"AP":1,"SSId":"TMO555","BSSId":"75:55:44:13:CC:DD","Channel":6,"Mode":"11n","RSSI":100,"Signal":-33,"LinkCount":1,"Downtime":"0T00:00:03"}}
16:10:20.882 RSL: STATE = {"Time":"2024-03-11T16:10:20","Uptime":"0T00:25:09","UptimeSec":1509,"Heap":163,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":45},"Wifi":{"AP":1,"SSId":"TMO555","BSSId":"75:55:44:13:CC:DD","Channel":6,"Mode":"11n","RSSI":100,"Signal":-35,"LinkCount":1,"Downtime":"0T00:00:03"}}
16:12:18.850 CMD: Weblog 4
16:12:18.853 RSL: RESULT = {"WebLog":4}
16:12:19.265 CFG: Saved, Count 18, Bytes 4096
16:12:36.373 WIF: Checking connection...
16:12:56.325 WIF: Checking connection...
16:13:16.327 WIF: Checking connection...
16:13:26.077 TCP: Got connection from 192.168.0.9
16:13:26.677 TCP: to MCU/1: FE00210120
16:13:26.726 TCP: from MCU: FE0261015F063B
16:13:26.777 TCP: to MCU/1: FE012749006F
16:13:26.827 TCP: from MCU: FE246749FEFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF0B
16:13:26.878 TCP: to MCU/1: FE052132010400000013
16:13:26.927 TCP: from MCU: FE03600002213272
16:13:26.978 TCP: Got connection from 192.168.0.9
16:13:26.980 TCP: to MCU/1: FE
16:13:27.578 TCP: to MCU/2: FE00210120
16:13:28.243 TCP: to MCU/2: EFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEF
16:13:28.258 TCP: to MCU/2: EF
16:13:30.679 TCP: to MCU/2: FE00210120
16:13:30.728 TCP: from MCU: FE0261015F063B
16:13:30.779 TCP: to MCU/2: FE012749006F
16:13:30.829 TCP: from MCU: FE246749FEFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF0B
16:13:30.879 TCP: to MCU/2: FE052132010400000013
16:13:30.928 TCP: from MCU: FE03600002213272
16:13:30.979 TCP: to MCU/2: FE022113210011
16:13:31.028 TCP: from MCU: FE026113740004
16:13:31.079 TCP: to MCU/2: FE04211C2100000018
16:13:31.130 TCP: from MCU: FE76611C0074AA05020B0F0B00640000000105018F000700020D1E0000001400000000000000000000001CBF0800008010020F0F0400010000000100000000A9A13D39A8DBFB27010000000000000000000000000000000000000000000000000000000000000000000000000F050000640A0100000000000000CB
16:13:31.179 TCP: to MCU/2: FE022113550065
16:13:31.228 TCP: from MCU: FE026113010071
16:13:31.308 TCP: to MCU/2: FE04211C550000006C
16:13:31.328 TCP: from MCU: FE03611C0001017E
16:13:31.379 TCP: to MCU/2: FE022113010031
16:13:31.428 TCP: from MCU: FE026113080078
16:13:31.479 TCP: to MCU/2: FE04211C0100000038
16:13:31.529 TCP: from MCU: FE0A611C0008F3472E2A004B120096
16:13:31.579 TCP: to MCU/2: FE0221138700B7
16:13:31.629 TCP: from MCU: FE026113010071
16:13:31.679 TCP: to MCU/2: FE04211C87000000BE
16:13:31.728 TCP: from MCU: FE03611C0001007F
16:13:31.779 TCP: to MCU/2: FE00210223
16:13:31.828 TCP: from MCU: FE0A6102020002071300000000007D
16:13:31.879 TCP: to MCU/2: FE0221133A000A
16:13:31.928 TCP: from MCU: FE026113110061
16:13:31.979 TCP: to MCU/2: FE04211C3A00000003
16:13:32.028 TCP: from MCU: FE02611C02007D
16:13:32.079 TCP: to MCU/2: FE
16:13:36.331 WIF: Checking connection...
16:13:56.331 WIF: Checking connection...
16:14:16.333 WIF: Checking connection...`

**Comment by Jason2866** [comment]
```
17:01:54.207 RUL: SYSTEM#BOOT performs "TCPStart 8888"
17:01:54.210 TCP: Starting TCP server on port 8888
17:01:54.213 RSL: RESULT = {"TCPStart":"Done"}
17:01:56.911 QPC: Reset
17:01:58.910 RSL: STATE = {"Time":"2024-03-10T17:01:58","Uptime":"0T00:00:08","UptimeSec":8,"Heap":161,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":45},"Wifi":{"AP":1,"SSId":"TMO555","BSSId":"75:55:44:13:CC:DD","Channel":6,"Mode":"11n","RSSI":100,"Signal":-36,"LinkCount":1,"Downtime":"0T00:00:03"}}
17:02:43.225 TCP: Got connection from 192.168.0.9
17:02:43.988 TCP: Got connection from 192.168.0.9
```
Connection established. Tasmota is not involved anymore from this point.

**Reply by tzago** [reply]
Hi Jason, thanks for the reply.
Actually connection is initiated but somehow not established. It is closed soon after ZHA from client side fails to negotiate the connection with Tasmoted Sonoff ZigbeePRO

**Comment by sfromis** [comment]
Looks like you did not make sure to also flash the firmware update for the CC2652?

**Reply by tzago** [reply]
Hi Sfromis,
Yeah this is the part where I dunno how to do it. I tried to use the webinstaller to flash the CC2652 but I see no option to do that.
Would you please let me know the how to process to flash my Sonoff HUB ZigbeePRO to this CC2652?
Should I use ESP-Flasher.exe to do it? 
What time we should flash the Sonoff ZigbeePRO with CC2652 before Tasmota or after Tasmota deployment using webinstaller?
And lastly where I can find the image CC2652 ?

**Reply by sfromis** [reply]
As mentioned in the instructions on the templates page you claimed to have been following:
> Once flashed, follow the coordinator firmware flashing Instructions from NotEnoughTech to flash the CC2652 chip itself from within Tasmota: [Flashing coordinator firmware by NotEnoughTech](https://notenoughtech.com/home-automation/tasmota-on-sonoff-zb-bridge-pro/#flash)

The files should already be in the file system, if you flashed the correct image.

**Reply by tzago** [reply]
Hi sfromis,

Thanks for prompt reply. 

Yes, trust me I followed this procedure from NotEnoughTech countless times and even by manual flash using ESP-Flasher.exe and downloading the tasmota tasmota32-zbbrdgpro.factory.bin manually flashing it
Also countless times flashing it via WebInstaller as well, including different versions to see if it worked but to no lucky.
Even tried to use the same Tasmota version included in the [Flashing coordinator firmware by NotEnoughTech](https://notenoughtech.com/home-automation/tasmota-on-sonoff-zb-bridge-pro/#flash) but looks like that image doesn't exist anymore Tasmota 11.1.0.3 


![image](https://github.com/arendst/Tasmota/assets/32423522/5299f73d-dea1-435c-af03-ab99598a6a82)


As you can see the image cc2652_flasher.be is there already.

What should I do from here then ?

**Reply by sfromis** [reply]
You may have a more basic issue. This is a pretty serious error message:
> `00:00:00.115 FRC: Some settings have been reset (2)`

That is when Tasmota was not able to boot successfully until starting to reset config settings. It is a recovery feature to not stay in a boot loop, but instead at least be able to start without the full config.

Possible reasons for this include not supplying enough power. Using USB to a computer is often not viable for getting ESP32 booted on Wifi.

**Reply by tzago** [reply]
Hi sfromis, 
Thanks again for the prompt reply.

Could you pleae clarify this point you mentioned above concerning the not sypplyiing enough power ?

Indeed when flashing it I only used the ttl usb provided all the power and data via the 5 pins attached to my Sonoff ZB Bridge Pro . I didn't attach the external power supply to my sonoff board via the micro usb port from the board and the flash process worked for both Web Installer and also via ESP-Flasher.exe.

Is this lack of power supply you're talking about by the moment we are flashing the Sonoff ZB Bridge Pro ?

After flashing it, I am using a regular iphone power supply brick to connect to my Sonoff ZB Bridge Pro.  Do we need a power brick more powerfull than a standard Iphone power supply brick?

**Reply by sfromis** [reply]
Normally, you can't boot ESP32 devices to Wifi via power from the USB/serial adapter used for flashing, as it cannot deliver enough power. After that, I'd expect a "charging" brick to be OK, unless unusually weak (rare nowadays).

But that was merely one possible cause, not me seeing it as a "diagnosis". And lacking a good crystal ball, I can't see what's happening....

**Reply by tzago** [reply]
![image](https://github.com/arendst/Tasmota/assets/32423522/7411f82d-1676-426e-8638-742523d114d0)

Tried to flash my Sonoff Zigbee Pro attaching all the 5 pins as usual with the usb ttl connected to my computer and with Power supply attached to its micso usb port and I get this message here.

**Reply by tzago** [reply]
I managed to hold the reset button and it flashed with power supply attached as well but the issue persists exactly like the opening issue statement for this discussion.
